ui/page/main_page.go

```diff
@@ -218,20 +231,51 @@ func (mp *MainPage) OnNavigatedTo() {
 			go mp.WL.MultiWallet.Politeia.Sync()
 		}
 	}
+
+	mp.updateExchangeSetting()
+
+	mp.UpdateBalance()
 }
 
 func (mp *MainPage) setLanguageSetting() {
 	langPre := mp.WL.Wallet.ReadStringConfigValueForKey(load.LanguagePreferenceKey)
 	values.SetUserLanguage(langPre)
 }
 
-func (mp *MainPage) UpdateBalance(fetchExch bool) {
-	if len(mp.totalBalanceUSD) == 0 || fetchExch {
-		load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
-	}
+func (mp *MainPage) updateExchangeSetting() {
 	currencyExchangeValue := mp.WL.Wallet.ReadStringConfigValueForKey(dcrlibwallet.CurrencyConversionConfigKey)
 	mp.usdExchangeSet = currencyExchangeValue == values.USDExchangeValue
+	if mp.usdExchangeSet {
+		go mp.fetchExchangeValue()
+	}
+}
+
+func (mp *MainPage) fetchExchangeValue() {
+	attempts, err := components.RetryFunc(components.RetryAttempts, 2*time.Second, func() error {
+		load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
+		if mp.dcrUsdtBittrex.LastTradeRate != "" {
+			log.Info("exchange value fetched")
+			mp.UpdateBalance()
+			mp.RefreshWindow()
+			mp.isLoadingUSDValue = false
+			return nil
+		}
+
+		mp.isLoadingUSDValue = true
+		return errors.New("failed to fetch dcrUsdtBittrex value")
+	})
+
+	if err != nil || mp.dcrUsdtBittrex.LastTradeRate == "" {
+		log.Error(err)
+
+		if attempts == components.RetryAttempts {
+			mp.isLoadingUSDValue = false
+		}
+		return
+	}
+}
```

Comment:
I initially had a few thoughts about this method, found it easier to just recommend an alternative snippet:
```
func (mp *MainPage) fetchExchangeValue() {
	if mp.isLoadingUSDValue {
		return
	}
	maxAttempts := 7
	delayBtwAttempts := 2 * time.Second
	mp.isLoadingUSDValue = true
	attempts, err := components.RetryFunc(maxAttempts, delayBtwAttempts, func() error {
		return load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
	})
	if err != nil {
		log.Errorf("error fetching usd exchange rate value after %d attempts: %v", attempts, err)
	} else if mp.dcrUsdtBittrex.LastTradeRate == "" {
		log.Errorf("no error while fetching usd exchange rate in %d tries, but no rate was fetched", attempts)
	} else {
		log.Infof("exchange rate value fetched: %s", mp.dcrUsdtBittrex.LastTradeRate)
		mp.UpdateBalance()
		mp.RefreshWindow()
	}
	mp.isLoadingUSDValue = false
}
```